modules/geo-location/class.jetpack-geo-location.php

```diff
@@ -0,0 +1,432 @@
+	}
+
+	/**
+	 * Output georss data for RSS items, assuming we have data for the currently rendered post and
+	 * that data as marked as public.
+	 */
+	public function rss_item() {
+		$meta_values = $this->get_meta_values( $this->get_post_id() );
+
+		if ( ! $meta_values['is_public'] ) {
+			return;
+		}
+
+		printf(
+			"\t<georss:point>%s %s</georss:point>\n",
+			ent2ncr( esc_html( $meta_values['latitude'] ) ),
+			ent2ncr( esc_html( $meta_values['longitude'] ) )
+		);
+
+		printf( "\t\t<geo:lat>%s</geo:lat>\n", ent2ncr( esc_html( $meta_values['latitude'] ) ) );
+		printf( "\t\t<geo:long>%s</geo:long>\n", ent2ncr( esc_html( $meta_values['longitude'] ) ) );
+	}
+
+	/**
+	 * Enqueue CSS for rendering post flair with geo-location.
+	 */
+	public function enqueue_scripts() {
+		wp_enqueue_style( 'dashicons' );
+	}
+
+	/**
+	 * If we're rendering a single post and public geo-location data is available for it,
+	 * include the human-friendly location label in the output.
+	 *
+	 * @param string $html
+	 *
+	 * @return string
+	 */
+	public function the_content_location_display( $content ) {
+		if (! $this->is_single() ) {
+			return $content;
+		}
+
+		return $content . $this->get_location_label();
+	}
+
+	/**
+	 * Get the HTML for displaying a label representing the location associated with the
+	 * supplied post ID.  If no post ID is given, we'll use the global $post variable, if
+	 * it is available.
+	 *
+	 * @param integer|null $post_id
+	 *
+	 * @return string
+	 */
+	public function get_location_label( $post_id = null ) {
+		$meta_values = $this->get_meta_values( $post_id ? $post_id : $this->get_post_id() );
+
+		if ( ! $meta_values['is_public'] ) {
+			return '';
+		}
+
+		// If the location has not been labeled, do not show the location
+		if ( ! $meta_values['label'] ) {
+			return '';
+		}
+
+		$html = '<div class="post-geo-location-label geo-chip">';
+		$html .= '<span class="dashicons dashicons-location" style="vertical-align: text-top;"></span> ';
+		$html .= esc_html( $meta_values['label'] );
+		$html .= '</div>';
+
+		/**
+		 * Allow modification or replacement of the default geo-location display HTML.
+		 *
+		 * @module geo-location
+		 *
+		 * @param array $html The default HTML for displaying a geo-location label.
+		 * @param array $geo_data An array containing "latitude", "longitude" and "label".
+		 */
+		$html = apply_filters( 'jetpack_geo_location_display', $html, $meta_values );
+
+		return $html;
+	}
+
+	/**
+	 * Get the ID of the current global post object, if available.  Otherwise, return null.
+	 *
+	 * This isolates the access of the global scope to this single method, making it easier to
+	 * safeguard against unexpected missing $post objects in other hook functions.
+	 *
+	 * @return int|null
+	 */
+	public function get_post_id() {
+		global $post;
+
+		if ( ! isset( $post) || ! $post || ! is_object( $post ) || ! isset( $post->ID ) ) {
```

Comment:
Do we need the `! $post` and `! is_object( $post )` here, or will `isset` suffice?

Reply:
Honestly, I'd rather be more paranoid than less in this case.  This code is run in the context of some pretty popular filters (the_content, wp_head, etc).  It's obviously a bug in a theme or a plugin if they've set the $post global to something other than a post object and then called apply_filters(), but if I can detect that and bail out early here, then I avoid geo-location features white-screening a site.  And because this get_post_id() method effectively isolates this paranoia to a single line of code, we're not really introducing any significant maintenance burden for future devs.